src/Persistence/MassTransit.Azure.Storage/AzureStorage/MessageData/AzureStorageMessageDataRepository.cs

```diff
@@ -115,7 +118,8 @@ public async Task<Stream> Get(Uri address, CancellationToken cancellationToken =
             {
                 LogContext.Debug?.Log("GET Message Data: {Address} ({Blob})", address, blob.Name);
 
-                return await blob.OpenReadAsync(new BlobOpenReadOptions(false), cancellationToken).ConfigureAwait(false);
+                var stream = await blob.OpenReadAsync(new BlobOpenReadOptions(false), cancellationToken).ConfigureAwait(false);
+                return blobName.EndsWith(".gz", StringComparison.OrdinalIgnoreCase) || _compress ? new GZipStream(stream, CompressionMode.Decompress, true) : stream;
```

Comment:
It seems like you're going to have some ownership/disposable issues with these streams to where they aren't closed and/or disposed properly.